### PR TITLE
[523]fix: define feature scenarios for collections

### PR DESCRIPTION
**Pull request by FenrirUnbound**
## Context

The team started working on the [homepage favorites](https://github.com/screwdriver-cd/screwdriver/issues/523) prior to creating a cucumber-js feature file. This feature file helps define scenarios that a user is expected to interact with the Screwdriver API and assert that the results are sane.

## Objective

Create a feature file for collections, asserting that the user can use the API in order to create, list, update, and delete collections.

## References

* Issue #523 

EDIT1: fixed link

### CODE REVIEW

File: features/collections.feature [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    Users want to be able to view a set of pipelines that they most care about. These
+    pipelines can be grouped into collections that these users own and maintain.
+
+    Collections can be modified to include and exclude pipelines over their lifetimes.
+    These collections can also be shared between users, such as teammates sharing
+    collections they're responsible for.
+
+    Rules:
+        - Collections are unique to an owning user and its name
```
Comment (r3rastogi) [inline features/collections.feature]: Grammatically, the "its" here is referring to the user. Maybe reword to "Collections are unique by name and owner"

**Comment by r3rastogi** [comment]
Why no "logged out" cases?

File: features/collections.feature [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        Then they can see that collection
+        And the collection contains that pipeline
+
+    @ignore
+    Scenario: Update Existing Collection
+        And "calvin" is logged in
+        When they create a new "myCollection"
+        Then they can see that collection
+        And the collection is empty
+        When they update the "myCollection" collection with the pipeline
```
Comment (r3rastogi) [inline features/collections.feature]: Give a specific pipeline number here

File: features/collections.feature [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    Scenario: Create New Collection
+        And "calvin" is logged in
+        When they create a new "myCollection" collection with that pipeline
+        Then they can see that collection
+        And the collection contains that pipeline
+
+    @ignore
+    Scenario: Update Existing Collection
+        And "calvin" is logged in
+        When they create a new "myCollection"
```
Comment (tkyi) [inline features/collections.feature]: can this either be "anotherCollection"
or just have it be assumed that they own the collection already
` And they have a collection "myCollection"`

File: features/collections.feature [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        Given an existing repository with these users and permissions:
+            | name          | permission  |
+            | calvin        | admin       |
+#            | miss wormwood | no access   |
+        And an existing pipeline with that repository
+
+    @ignore
+    Scenario: Create New Collection
+        And "calvin" is logged in
+        When they create a new "myCollection" collection with that pipeline
```
Comment (tkyi) [inline features/collections.feature]: a pipeline

Comment (tkyi) [inline features/collections.feature]: Oh okay I see that it refers to the background pipeline

File: features/collections.feature (hunk 3 above)
Comment (FenrirUnbound) [inline features/collections.feature]: That's so we can re-use step definitions

Comment (tkyi) [inline features/collections.feature]: You can probably say "When they create a new collection __________"

File: features/collections.feature (hunk 2 above)
Comment (tkyi) [inline features/collections.feature]: Can you make it consistent by putting "collection" before "nameOfCollection" here, like in lines 41 and 42.

File: features/collections.feature (hunk 3 above)
Comment (tkyi) [inline features/collections.feature]: Still need to add the word "collection" here

File: features/collections.feature [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    @ignore
+    Scenario: Update Existing Collection
+        And "calvin" is logged in
+        When they create a new "myCollection"
+        Then they can see that collection
+        And the collection is empty
+        When they update the "myCollection" collection with the pipeline
+        Then they can see that collection
+        And the collection contains that pipeline
```
Comment (d2lam) [inline features/collections.feature]: Seems like just 1 pipeline in this case? We should also test for multiple pipelines case.

File: features/collections.feature [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        And they have a collection "anotherCollection"
+        When they fetch all their collections
+        Then they can see those collections
+
+    @ignore
+    Scenario: Deleting A Collection
+        And "calvin" is logged in
+        And they have a collection "myCollection"
+        When they delete that collection
+        Then that collection no longer exists
```
Comment (d2lam) [inline features/collections.feature]: We should add a line here to make sure the collection is deleted but the pipelines remain.

File: features/collections.feature [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    Users want to be able to view a set of pipelines that they most care about. These
+    pipelines can be grouped into collections that these users own and maintain.
+
+    Collections can be modified to include and exclude pipelines over their lifetimes.
+    These collections can also be shared between users, such as teammates sharing
+    collections they're responsible for.
+
+    Rules:
+        - Collections are unique by name and owner
```
Comment (d2lam) [inline features/collections.feature]: Should add a test for the same user creating a collection with the same name should give an error

File: features/collections.feature (hunk 5 above)
Comment (FenrirUnbound) [inline features/collections.feature]: It's meant to cover the case where a user initializes a collection with a pipeline. Do you want it to assert that it can be initialized with more than one pipeline?

File: features/collections.feature (hunk 6 above)
Comment (FenrirUnbound) [inline features/collections.feature]: > We should add a line here to make sure the collection is deleted but the pipelines remain.

Collections have nothing to do with maintaining the lifespan of pipelines. Why do we want to check that the pipelines are unaffected?

**Comment by FenrirUnbound** [comment]
> Why no "logged out" or "no access" cases?

This is intended for certifying core functionality, so these cases will be addressed in a future update.

File: features/collections.feature (hunk 5 above)
Comment (d2lam) [inline features/collections.feature]: @FenrirUnbound  Yes

File: features/collections.feature (hunk 6 above)
Comment (d2lam) [inline features/collections.feature]: I think the expected behaviors of deleting collections is the pipelines are not deleted, so it's safer to assert that here. If you don't want it, it's ok.

Comment (FenrirUnbound) [inline features/collections.feature]: > I think the expected behaviors of deleting collections is the pipelines are not deleted, so it's safer to assert that here. 

Added the assertion 👍
